src/platform/msvc/dirent.h

```diff
@@ -0,0 +1,917 @@
+/*
```

Comment:
Please write a VDir for Win32 instead of using a dirent shim. The whole point of VDir is to have platform-specific implementations instead of having to reimplement all of dirent each time.